pkg/scalers/redis_scaler.go

```diff
@@ -164,7 +164,7 @@ func parseRedisMetadata(config *ScalerConfig, parserFn redisAddressParser) (*red
 
 	meta.databaseIndex = defaultDBIdx
 	if val, ok := config.TriggerMetadata["databaseIndex"]; ok {
-		dbIndex, err := strconv.ParseInt(val, 10, 64)
+		dbIndex, err := strconv.ParseInt(val, 10, 32)
```

Comment:
as per https://stackoverflow.com/questions/36735236/maximum-number-of-databases-in-redis this change should be ok.